table/block_based/block.h

```diff
@@ -150,24 +152,70 @@ class BlockReadAmpBitmap {
 // for details of the format and the various block types.
 class Block {
  public:
-  // Initialize the block with the specified contents.
-  explicit Block(BlockContents&& contents, size_t read_amp_bytes_per_bit = 0,
-                 Statistics* statistics = nullptr);
   // No copying allowed
   Block(const Block&) = delete;
   void operator=(const Block&) = delete;
 
-  ~Block();
+  virtual ~Block();
 
-  size_t size() const { return size_; }
   const char* data() const { return data_; }
+  uint32_t block_size() const {
```

Comment:
For small functions defined in headers, I suggest marking `inline`. For us this means "I don't need to distinguish this function from its caller for profiling purposes" and potentially saves some accounting overhead.